pkg/trace/api/evp_intake_proxy_test.go

```diff
@@ -0,0 +1,333 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/trace/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type roundTripperMock func(*http.Request) (*http.Response, error)
+
+func (r roundTripperMock) RoundTrip(req *http.Request) (*http.Response, error) {
+	return r(req)
+}
+
+// sendRequestThroughForwarder sends a request through the evpProxyForwarder handler and returns the forwarded
+// request(s), their response and the log output. The path for inReq shouldn't have the /evp_proxy/v1
+// prefix since it is passed directly to the inner proxy handler and not the trace-agent API handler.
+func sendRequestThroughForwarder(conf *config.AgentConfig, inReq *http.Request) (outReqs []*http.Request, resp *http.Response, logs string) {
+	mockRoundTripper := roundTripperMock(func(req *http.Request) (*http.Response, error) {
+		if req.Body != nil {
+			if _, err := ioutil.ReadAll(req.Body); err != nil && err != io.EOF {
+				return nil, err
+			}
+		}
+		outReqs = append(outReqs, req)
+		// If we got here it means the proxy didn't raise an error earlier, return an ok resp
+		return &http.Response{
+			StatusCode: 200,
+			Body:       ioutil.NopCloser(bytes.NewBuffer([]byte("ok_resprino"))),
+		}, nil
+	})
+	var loggerBuffer bytes.Buffer
+	mockLogger := log.New(io.Writer(&loggerBuffer), "", 0)
+	handler := evpProxyForwarder(conf, evpProxyEndpointsFromConfig(conf), mockRoundTripper, mockLogger)
+	rec := httptest.NewRecorder()
+	handler.ServeHTTP(rec, inReq)
+	return outReqs, rec.Result(), loggerBuffer.String()
+}
+
+func TestEVPProxyForwarder(t *testing.T) {
+
+	randBodyBuf := make([]byte, 1024)
```

Comment:
```suggestion
	buf := make([]byte, 1024)
```
Should be enough :)

Reply:
This is used far away from its definition, I think it's better to specify what this buffer is for.